src/Raven.Server/ServerWide/Commands/PutCertificateWithSamePinningHashCommand.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using Raven.Client.ServerWide.Operations.Certificates;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.ServerWide.Commands
+{
+    public class PutCertificateWithSamePinningHashCommand : PutCertificateCommand
+    {
+        public PutCertificateWithSamePinningHashCommand()
+        {
+            // for deserialization
+        }
+
+        public PutCertificateWithSamePinningHashCommand(string name, CertificateDefinition value) : base(name, value)
+        {
+
+        }
+
+        public override DynamicJsonValue ValueToJson()
+        {
+            return Value?.ToJson();
+        }
+
+        public override void VerifyCanExecuteCommand(ServerStore store, TransactionOperationContext context, bool isClusterAdmin)
```

Comment:
This is only valid for admin callers.
This may be _generated_ by any user, but the command is sent by the cluster itself and require cluster admin permissions

Reply:
done